### ISSUE TITLE
Bootstrapping RHEL clones fail when using a proxy

### ISSUE THREAD

**Issue by nodeg** [uyuni-project/uyuni#7442]
### Problem description

I cannot bootstrap RHEL clones (AlmaLinux 9, Rocky Linux 9/8, Oracle Linux 9, CentOS 7) with a proxy. When using the server, it works. What I am wondering about is the fact that it should also not work without proxy because of the wrong folder structure.
According to the error message when using a proxy, it expects the bootstrap files for e.g. AlmaLinux 9 in
```
/srv/www/htdocs/pub/repositories/res/9/bootstrap
```
but the files are created under
```
/srv/www/htdocs/pub/repositories/almalinux/9/bootstrap
```
with `mgr-create-bootstrap-repo`:

```bash
# uyuni-XXXX:~ # mgr-create-bootstrap-repo
1. SLE-12-SP4-x86_64
2. SLE-12-SP5-x86_64
3. SLE-15-SP1-x86_64
4. SLE-15-SP2-x86_64
5. SLE-15-SP3-x86_64
6. SLE-15-SP4-x86_64
7. SLE-15-SP5-x86_64
8. SLE-MICRO-5.1-x86_64
9. SLE-MICRO-5.2-x86_64
10. SLE-MICRO-5.3-x86_64
11. SUSE-LibertyLinux9-x86_64
12. almalinux-9-x86_64-uyuni
13. centos-7-x86_64-uyuni
14. debian10-amd64-uyuni
15. debian11-amd64-uyuni
16. openSUSE-Leap-15.4-x86_64-uyuni
17. oracle-9-x86_64-uyuni
18. rockylinux-8-x86_64-uyuni
19. rockylinux-9-x86_64-uyuni
20. ubuntu-18.04-amd64-uyuni
21. ubuntu-20.04-amd64-uyuni
22. ubuntu-22.04-amd64-uyuni
Enter a number of a product label: 12

Creating bootstrap repo for almalinux-9-x86_64-uyuni
copy 'venv-salt-minion-3004-14.24.uyuni.x86_64'
Directory walk started
Directory walk done - 1 packages
Temporary output repo path: /srv/www/htdocs/pub/repositories/almalinux/9/bootstrap.tmp/.repodata/
Preparing sqlite DBs
Pool started (with 5 workers)
Pool finished
```

## Bootstrap directory structure on the server
```bash
uyuni-bv-master-srv:/srv/www/htdocs/pub/repositories # tree -L 4
.
├── almalinux
│   └── 9
│       └── bootstrap
│           ├── repodata
│           ├── venv-enabled-x86_64.txt
│           └── x86_64
├── centos
│   └── 7
│       └── bootstrap
│           ├── repodata
│           ├── venv-enabled-x86_64.txt
│           └── x86_64
├── debian
│   ├── 10
│   │   └── bootstrap
│   │       ├── conf
│   │       ├── db
│   │       ├── dists
│   │       ├── pool
│   │       ├── venv-enabled-amd64.txt
│   │       └── venv-enabled-i386.txt
│   └── 11
│       └── bootstrap
│           ├── conf
│           ├── db
│           ├── dists
│           ├── pool
│           ├── venv-enabled-amd64.txt
│           └── venv-enabled-i386.txt
├── empty
│   └── repodata
│       ├── filelists.xml.gz
│       ├── other.xml.gz
│       ├── primary.xml.gz
│       └── repomd.xml
├── empty-deb
│   ├── Packages
│   └── Release
├── opensuse
│   └── 15
│       └── 4
│           └── bootstrap
├── oracle
│   └── 9
│       └── bootstrap
│           ├── repodata
│           ├── venv-enabled-x86_64.txt
│           └── x86_64
├── res
│   └── 9
│       └── bootstrap
│           └── repodata
├── rockylinux
│   ├── 8
│   │   └── bootstrap
│   │       ├── noarch
│   │       ├── repodata
│   │       ├── venv-enabled-x86_64.txt
│   │       └── x86_64
│   └── 9
│       └── bootstrap
│           ├── repodata
│           ├── venv-enabled-x86_64.txt
│           └── x86_64
├── sle
│   ├── 12
│   │   ├── 4
│   │   │   └── bootstrap
│   │   └── 5
│   │       └── bootstrap
│   └── 15
│       ├── 1
│       │   └── bootstrap
│       ├── 2
│       │   └── bootstrap
│       ├── 3
│       │   └── bootstrap
│       ├── 4
│       │   └── bootstrap
│       └── 5
│           └── bootstrap
├── slemicro
│   └── 5
│       ├── 1
│       │   └── bootstrap
│       ├── 2
│       │   └── bootstrap
│       └── 3
│           └── bootstrap
└── ubuntu
    ├── 18
    │   └── 4
    │       └── bootstrap
    ├── 20
    │   └── 4
    │       └── bootstrap
    └── 22
        └── 4
            └── bootstrap

84 directories, 15 files
```
![image](https://github.com/uyuni-project/uyuni/assets/12104291/242b856f-e378-47f5-8399-ca8c8c309962)

The same applies to Rocky Linux 9/8, Oracle Linux 9 and CentOS 7

### Rocky 8
```bash
Error: Unable to download https://uyuni-XXXXXXXXX:443/pub/repositories/res/8/bootstrap/venv-enabled-x86_64.txt file!
```

### Rocky 9
```bash
Error: Unable to download https://uyuni-XXXXXXXXX:443/pub/repositories/res/9/bootstrap/venv-enabled-x86_64.txt file!
```

### CentOS 7

```bash
Error: Unable to download https://uyuni-XXXXXXXXX:443/pub/repositories/res/7/bootstrap/venv-enabled-x86_64.txt file!
```

### Oracle 9

```bash
Error: Unable to download https://uyuni-XXXXXXXXX:443/pub/repositories/res/9/bootstrap/venv-enabled-x86_64.txt file!
```

### Steps to reproduce

1. Add and synchronize the RHEL clone channels with `spacewalk-common-channels` and create an activation key for the minion
2. Create bootstrap repositories with `mgr-create-bootstrap-repos` for it
3. Onboard the minion with a proxy through the webUI
4. See the error message
5. Onboard the minion without the proxy and see that it worked

### Uyuni version

```bash
uyuni-bv-master-srv:~ # zypper info Uyuni-Server-release
Loading repository data...
Reading installed packages...


Information for package Uyuni-Server-release:
---------------------------------------------
Repository     : server_pool_repo
Name           : Uyuni-Server-release
Version        : 2023.04-220400.204.2.uyuni2
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes (automatically)
Status         : up-to-date
Source package : Uyuni-Server-release-2023.04-220400.204.2.uyuni2.src
Summary        : Uyuni Server
```

### Uyuni proxy version (if used)

```bash
uyuni-bv-master-pxy:~ # zypper info Uyuni-Proxy-release
Loading repository data...
Reading installed packages...


Information for package Uyuni-Proxy-release:
--------------------------------------------
Repository     : Uyuni Proxy Stable for openSUSE Leap 15.4 (x86_64)
Name           : Uyuni-Proxy-release
Version        : 2023.04-220400.204.2.uyuni2
Arch           : x86_64
Vendor         : obs://build.opensuse.org/systemsmanagement:Uyuni
Support Level  : Level 3
Installed Size : 1.4 KiB
Installed      : Yes (automatically)
Status         : up-to-date
Source package : Uyuni-Proxy-release-2023.04-220400.204.2.uyuni2.src
Summary        : Uyuni Proxy
```

**Comment by mcalmer** [comment]
@nodeg I just tried with the latest Uyuni and I cannot reproduce the problem. Maybe it was fixed in the meanwhile.

We first test the the rocky URL against the proxy:
```
"GET /pub/repositories/rockylinux/9/bootstrap/venv-enabled-x86_64.txt HTTP/1.1" 200 120 "-" "Salt/3006.0 http.query()"
```

and it is successful (200 OK). After that it download venv-salt-minion package and does the bootstrap.
In case that URL on the proxy would return a 404 Not found or another error, it would fallback to "res".

Feel free and try to reproduce it, but from my point of view things are working now.
